### PR TITLE
Add middleware injection for dev server & `.makeRequest()`

### PR DESCRIPTION
**Pull request by codetheweb**

### CODE REVIEW

**Comment by codetheweb** [comment]
turns out the scuffed commit history was because I squash merged the first PR in a chain that depended on each other 😅

File: src/dev/headless/request-handler-controller.ts [hunk 1]
```diff
@@ -52,8 +54,13 @@ export class RequestHandlerController {
           await this.bundlePathPromise,
           "utf-8"
         )
+        const { middlewares } = this
         this.cachedWinterCGRuntime = new EdgeRuntime({
           initialCode: contents,
+          extend(context) {
+            context._injectedEdgeSpecMiddlewares = middlewares
+            return context
+          },
```
Comment (mxsdev) [inline src/dev/headless/request-handler-controller.ts:64]: Why can't we wrap the middleware at top level?

Comment (codetheweb) [reply src/dev/headless/request-handler-controller.ts:64]: sorry, what do you mean by top level?

my plan is to replace this PR by bundling a module service entrypoint by default, then allowing that entrypoint to accept middleware (probably will still have to inject using this method for the edge runtime)

**Review by mxsdev — APPROVED** [review]
dope 👍

**Pull request merged**